lib/commentOnOpenedPullRequest.js

```diff
@@ -0,0 +1,128 @@
+ * @param {String} pullRequestCommentsUrl API Url to POST comments to
+ * @param {Object} headers Headers
+ * @param {String[]} thirdPartyFolders Folder paths to search for third party file changes
+ * @private
+ * @return {Promise<http.IncomingMessage | String>} Response of POST
+ */
+commentOnOpenedPullRequest._implementation = function (pullRequestFilesUrl, pullRequestCommentsUrl, headers, thirdPartyFolders) {
+    return requestPromise.get({
+        url: pullRequestFilesUrl,
+        headers: headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+    .then(function (filesJsonResponse) {
+        return checkStatus(filesJsonResponse);
+    })
+    .then(function (filesJsonResponse) {
+        var message = '';
+        var files;
+        var didUpdateChanges;
+        var didUpdateThirdParty;
+
+        files = filesJsonResponse.body.map(function (file) {
+            return file.filename;
+        });
+        dateLog('These files changed in the pull request: ' + files);
+
+        didUpdateChanges = commentOnOpenedPullRequest._didUpdateChanges(files);
+        didUpdateThirdParty = commentOnOpenedPullRequest._didUpdateThirdParty(files, thirdPartyFolders);
+        if (didUpdateChanges) {
+            dateLog('CHANGES.md was updated');
+            message += changesMessage;
+        }
+        if (didUpdateThirdParty) {
+            dateLog('A third party file changed');
+            message += '\nAlso: ' + thirdPartyMessage;
+        }
+        if (didUpdateChanges || didUpdateThirdParty) {
+            return requestPromise.post({
+                uri: pullRequestCommentsUrl,
+                headers: headers,
+                body: {
+                    body: message
+                },
+                json: true,
+                resolveWithFullResponse: true
+            });
+        }
+        return Promise.resolve();
+    });
+};
+
+/** Check if CHANGES.md updated
+ *
+ * @param {String[]} files Paths to changed files
+ * @private
+ * @return {boolean} True `CHANGES.md` has changed
+ */
+commentOnOpenedPullRequest._didUpdateChanges = function (files) {
+    for (var i = 0; i < files.length; i++) {
+        if (/^CHANGES\.md/.test(files[i])) {
+            return true;
+        }
+    }
+    return false;
+};
+
+/** Check if any third party files were updated
+ *
+ * @param {String[]} files Paths to changed files
+ * @param {String[]} thirdPartyFolders Paths to third party folders
+ * @private
+ * @return {boolean} True if any file in `thirdPartyFolders` has changed
+ */
+commentOnOpenedPullRequest._didUpdateThirdParty = function (files, thirdPartyFolders) {
+    if (thirdPartyFolders.length === 0) {
+        return false;
+    }
+    for (var i = 0; i < files.length; i++) {
+        for (var j = 0; j < thirdPartyFolders.length; j++) {
+            console.log(files[i], thirdPartyFolders[j]);
+            if (files[i].startsWith(thirdPartyFolders[j]) || ('/' + files[i]).startsWith(thirdPartyFolders[j])) {
```

Comment:
In case, when specifying the `ThirdParty` folders, someone puts a `/` in front. Like, this would not match:

`[/ThirdParty/]`, but this would: `[ThirdParty/]`